### PR TITLE
New page to document Asset Processor and Asset Builder Settings

### PR DESCRIPTION
**Pull request by lmbr-pip**
## Change summary

New page to document Asset Processor and Asset Builder Settings.
* Also fixes a few spelling mistakes/typos on a related pages.

Can break into two, maybe move the AB settings to the debug AB settings section of the debugging guide?

### Submission Checklist:

* [x] **Descriptive active voice** - Do descriptive sentences have a clear *subject* and *action verb*?
* [x] **Answer the question at hand** - Does the documentation answer a *what*, *why*, *how*, or *where* type of question?
* [x] **Consistency** - Does the content consistently follow the [style guide](https://o3de.org/docs/contributing/to-docs/style-guide/)?
* [x] **Help the user** - Does the documentation show the user something *meaningful*?

### CODE REVIEW

**Comment by lmbr-pip** [comment]
I need to reset my main history it seems. Depending on PR feedback, I may recreate this PR on a cleaner branch.

File: content/docs/user-guide/assets/asset-processor/settings.md [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+| waitOnLaunch                    | Briefly pauses Asset Processor during initializiation. Useful if you want to attach a debugger.                                                                                                     |
+| zeroAnalysisMode                | Enables using file modification time when examining source assets for processing.                                                                                                                   | 
+| enableQueryLogging              | Enables logging database queries.                                                                                                                                                                   |
+| dependencyScanPattern (dsp)     | Scans assets that match the given pattern for missing product dependencies.                                                                                                                         |
+| fileDependencyScanPattern (fsp) | Used with `dependencyScanPattern` to farther filter the scan                                                                                                                                        |
```
Comment (AMZN-stankowi) [inline content/docs/user-guide/assets/asset-processor/settings.md:30]: nitpick: You're using a period at the end of the other descriptions, but not here

**Review by AMZN-stankowi — APPROVED** [review]

File: content/docs/user-guide/assets/asset-processor/settings.md (hunk 1 above)
Comment (lmbr-pip) [reply content/docs/user-guide/assets/asset-processor/settings.md:30]: Will fix

File: content/docs/user-guide/assets/asset-processor/settings.md [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+| engine-path | Supplies the engine path to the engine.                             |
+
+You can also control the behavior of Asset Processor or Asset Processor Batch using these settings:
+
+| Setting                         | Description                                                                                                                                                                                         |
+|---------------------------------|-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| waitOnLaunch                    | Briefly pauses Asset Processor during initializiation. Useful if you want to attach a debugger.                                                                                                     |
+| zeroAnalysisMode                | Enables using file modification time when examining source assets for processing.                                                                                                                   |
```
[the remark this answers is not in the document]
Comment (lmbr-pip) [reply content/docs/user-guide/assets/asset-processor/settings.md:27]: I can try and dig through the code, but unlike CVARs the defaults are not clearly defined, but set in disconnected code.

File: content/docs/user-guide/assets/asset-processor/_index.md [hunk 3]
```diff
@@ -1,45 +1,46 @@
...
+
+| Topic | Description |
+| - | - |
+| [Interface](interface) | Learn to navigate Asset Processor's interface, and how to read the status of process jobs. |
+| [Asset Processor Batch](asset-processor-batch) | Use Asset Processor batch to batch process the source assets for a project in an automated build system. |
+| [Configuration](configuration) | Configure Asset Processor by setting options in the `AssetProcessorPlatformConfig.setreg` configuration file. |
+| [Faster Scanning](faster-scanning) | Learn how Asset processor determines when assets should be processed and how to set Asset Processor's scanning mode. |
+| [Debugging](debugging) | Learn several methods you can use to debug Asset Processor issues. |
+| [Move Assets](move-assets) | Learn how to move assets to new directory locations in O3DE while maintaining internal references. |
+| [Settings](settibgs) | Learn how to use advance settings to control Asset Processor and Asset Processor Batch. |
```
Comment (jackalbe) [inline content/docs/user-guide/assets/asset-processor/_index.md:46]: settibgs?

File: content/docs/user-guide/assets/asset-processor/settings.md [hunk 4]
```diff
@@ -0,0 +1,41 @@
+---
+linkTitle: Asset Processor Settings
+title: Asset Processor Settings
+description: Commandline settings for AssetProcessor and AssetBuilder
+weight: 900
+toc: true
+---
+
+The Asset Processor and Asset Processor Batch expose a number of commandline settings that can be used when manually launching.
```
Comment (jackalbe) [inline content/docs/user-guide/assets/asset-processor/settings.md:9]: maybe a command line example of usage?

File: content/docs/user-guide/assets/asset-processor/settings.md [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+| project-path       | Supplies the path to the project that the AP should use (required). | 
+| engine-path        | Supplies the engine path to the engine.                             |
+
+
+## Advanced Settings
+You can also control the behavior using the following command line options:
+
+| Setting                         | Description                                                                                                                                                                                                 | Notes                                    |
+|---------------------------------|-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|------------------------------------------|
+| waitOnLaunch                    | Briefly pauses Asset Processor during initialization. Useful if you want to attach a debugger.                                                                                                              | Defaults to false for AP, true for batch |
```
Comment (amzn-mike) [inline content/docs/user-guide/assets/asset-processor/settings.md:27]: The comment says this is false for AP and true for batch by default.  Is that true?  I don't think APB normally pauses to wait for a debugger

File: content/docs/user-guide/assets/asset-processor/settings.md [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+| zeroAnalysisMode                | Enables using file modification time when examining source assets for processing.                                                                                                                           |                                          |
```
Comment (amzn-mike) [inline content/docs/user-guide/assets/asset-processor/settings.md:28]: This is true by default for AP GUI and false by default for AP Batch

File: content/docs/user-guide/assets/asset-processor/settings.md [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+| enableQueryLogging              | Enables logging database queries.                                                                                                                                                                           |                                          |
+| dependencyScanPattern (dsp)     | Scans assets that match the given pattern for missing product dependencies.                                                                                                                                 |                                          |
+| fileDependencyScanPattern (fsp) | Used with `dependencyScanPattern` to farther filter the scan                                                                                                                                                |                                          |
+| additionalScanFolders.          | Used with `dependencyScanPattern` to farther filter the scan.                                                                                                                                               |                                          |
```
Comment (amzn-mike) [inline content/docs/user-guide/assets/asset-processor/settings.md:32]: There probably shouldn't be a period included at the end of the argument name

File: content/docs/user-guide/assets/asset-processor/_index.md (hunk 3 above)
Comment (micronAMZN) [inline content/docs/user-guide/assets/asset-processor/_index.md:46]:
```suggestion
| [Settings](settings) | Learn how to use advance settings to control Asset Processor and Asset Processor Batch. |
```

File: content/docs/user-guide/assets/asset-processor/debugging.md [hunk 8]
```diff
@@ -1,19 +1,19 @@
...
 If you are having issues with **Asset Processor**, this page covers common problems, solutions, and debugging techniques.
 
 Some of the most common problems we've seen with Asset Processor include:
 
-* After changing a source asset, it takes too long for assets to finish processing.
-* The product assets don't behave like expected. They may be missing entirely, missing information, have incorrect information.
-* Unexpected warnings or errors.
-* Asset packaging isn't working as expected, content is missing or content is being included in bundles that was not expected to be there.
+* After changing a source asset, it takes too long for assets to finish processing.
+* The product assets don't behave like expected. They may be missing entirely, missing information, have incorrect information.
```
Comment (micronAMZN) [inline content/docs/user-guide/assets/asset-processor/debugging.md:14]:
```suggestion
* Asset processing takes too long after changes are made to a source asset.
* The product assets don't behave like expected. They may have incorrect information, be missing information, or be missing entirely.
```

File: content/docs/user-guide/assets/asset-processor/debugging.md [hunk 9]
```diff
@@ -1,19 +1,19 @@
...
+* Unexpected warnings or errors.
+* Asset packaging isn't working as expected, content is missing or content is being included in bundles that was not expected to be there.
```
Comment (micronAMZN) [inline content/docs/user-guide/assets/asset-processor/debugging.md:16]:
```suggestion
* Asset packaging isn't working as expected. Content is missing or content is being included in bundles that was not expected to be there.
```

File: content/docs/user-guide/assets/pipeline/asset-builders.md [hunk 10]
```diff
@@ -1,40 +1,69 @@
...
+| debug_create       | Debug mode for the create job of the specified file.                                                                                                                                                          |
+| debug_process      | Debug mode for the process job of the specified file.                                                                                                                                                         |
+| tags               | Additional tags to add to the debug platform for job processing. One tag can be supplied per option.                                                                                                          |
+| platform           | Platform to use for debugging. ex: pc                                                                                                                                                                         |
```
Comment (micronAMZN) [inline content/docs/user-guide/assets/pipeline/asset-builders.md:69]:
```suggestion
| platform           | Platform to use for debugging, for example, `pc` |
```

File: content/docs/user-guide/assets/asset-processor/settings.md [hunk 11]
```diff
@@ -0,0 +1,41 @@
+---
+linkTitle: Asset Processor Settings
```
Comment (micronAMZN) [inline content/docs/user-guide/assets/asset-processor/settings.md:2]: Shortening link title for the ToC to match other AP topics.

```suggestion
linkTitle: Settings
```

File: content/docs/user-guide/assets/asset-processor/settings.md (hunk 5 above)
Comment (lmbr-pip) [reply content/docs/user-guide/assets/asset-processor/settings.md:27]: Opps, wrong line. Am still auditing AP vs AP Batch changes. Will update.

File: content/docs/user-guide/assets/asset-processor/settings.md (hunk 6 above)
Comment (lmbr-pip) [reply content/docs/user-guide/assets/asset-processor/settings.md:28]: Thanks will add.

File: content/docs/user-guide/assets/asset-processor/settings.md (hunk 7 above)
Comment (lmbr-pip) [reply content/docs/user-guide/assets/asset-processor/settings.md:32]: Good call, will fix.

File: content/docs/user-guide/assets/asset-processor/settings.md (hunk 4 above)
Comment (lmbr-pip) [reply content/docs/user-guide/assets/asset-processor/settings.md:9]: Good suggestion, will add

**Comment by c5h11oh** [comment]
Would it be helpful if we add instructions on how to pass in those settings/switches? In particular, add some descriptions like
> You can enable settings by appending `--<setting>[ value]`, `/<setting>[ value]`  or `/<setting>[=value]` after the program to be launched. For example, `AssetProcessor.exe /warningLevel=2`.

The parsing is a generic tool under AzTool though, so I am not sure if this has already been documented somewhere else that I didn't notice. I learned this from discovering the command line parsing tool header:
(copied from CommandLine.h)
```
    /**
    * Given a commandline, this allows uniform parsing of it into parameter values and extra values.
    * When parsed, the commandline becomes a series of "switches" or "extra values"
    * For example, a switch may be specified as either 
    *      /switchName=value1[,value2...]
    *      /switchname value[,value2...]
    *      --switchname value[,value2...]
    *    or other combinations of the above
    * You may NOT use a colon as a switch separator since file names may easily contain them.
    * any additional parameters which are not associated with any switch are considered "misc values"
    */
```

**Comment by micronAMZN** [comment]
Can we resolve the open conversations and move forward with this PR?

**Comment by lmbr-pip** [comment]
My apologies, have been OOTO and swamped with other work. Will try and address all feedback by end of week (July 22)

File: content/docs/user-guide/assets/asset-processor/settings.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (lmbr-pip) [reply content/docs/user-guide/assets/asset-processor/settings.md]: Added note

File: content/docs/user-guide/assets/asset-processor/settings.md [hunk 12]
```diff
@@ -0,0 +1,46 @@
...
+| fileDependencyScanPattern (fsp) | Used with `dependencyScanPattern` to farther filter the scan.                                                                                                                                               |                                                                                                 |
+| help (h)                        | Displays all available commands and their description.                                                                                                                                                      |                                                                                                 |
+| sortJobsByDBSourceName          | When enabled, sorts pending jobs with equal priority and dependencies by database source name instead of job ID. Useful for automated tests to process assets in the same order each time.                  |                                                                                                 |
```
Comment (AMZN-stankowi) [inline content/docs/user-guide/assets/asset-processor/settings.md:40]: this is the default behavior now, and this flag has been removed

**Review by AMZN-stankowi — APPROVED** [review]

File: content/docs/user-guide/assets/pipeline/asset-builders.md [hunk 13]
```diff
@@ -1,40 +1,69 @@
...
+
+`ProcessJobResponse` places the product assets in a temp directory. Asset Processor moves the product assets to the **Asset Cache** and populates the **Asset Database** and **Asset Catalog** with information it uses to track the product assets, product dependencies, and the jobs that produced them.
+
+Process Job is multithreaded. Several Asset Builders can run multiple process jobs simultaneously.
+
+
+### Settings
+You can run Asset Builder in debug mode when develop new features for a builder or to debug issues in the asset pipeline. In debug mode, Asset Builder creates a test job or processes jobs for specified files.
+
+> You must start Asset Processor before you can enter a -debug command.
```
Comment (jackalbe) [inline content/docs/user-guide/assets/pipeline/asset-builders.md:46]: maybe use:
{{< note >}}
You must start Asset Processor before you can enter a -debug command.
{{< /note >}}
?

**Review by jackalbe — APPROVED** [review]

File: content/docs/user-guide/assets/asset-processor/settings.md (hunk 12 above)
Comment (lmbr-pip) [reply content/docs/user-guide/assets/asset-processor/settings.md:40]: @AMZN-stankowi Note this is targetting the 2205 release / main and not development.

File: content/docs/user-guide/assets/asset-processor/settings.md [hunk 14]
```diff
@@ -8,34 +8,39 @@ toc: true
...
+| dependencyScanPattern (dsp)     | Scans assets that match the given pattern for missing product dependencies.                                                                                                                                 |                                                                                                 |
+| enableQueryLogging              | Enables logging database queries.                                                                                                                                                                           |                                                                                                 |
+| fileDependencyScanPattern (fsp) | Used with `dependencyScanPattern` to farther filter the scan.                                                                                                                                               |                                                                                                 |
```
Comment (amzn-mike) [inline content/docs/user-guide/assets/asset-processor/settings.md:38]: I think this should be "further" instead of "farther"

**Review by amzn-mike — APPROVED** [review]

File: content/docs/user-guide/assets/asset-processor/settings.md [hunk 15]
```diff
@@ -0,0 +1,46 @@
...
+| project-name       | Name of the current project.                                        | On Windows, to the current project's name             |
+| project-cache-path | Full path to the project cache folder.                              | On Windows, to the current project's cache path       |
+| project-path       | Supplies the path to the project that the AP should use (required). | On Windows, to the current project's path             |                                  
+| engine-path        | Supplies the engine path to the engine.                             | On Windows, to the engine registered with the project |
+
+
+## Advanced settings
+You can also control the behavior using the following command line options:
+
+| Setting                         | Description                                                                                                                                                                                                 | Notes                                                                                           |
```
Comment (c5h11oh) [inline content/docs/user-guide/assets/asset-processor/settings.md:30]: I think the "Notes" can be renamed as "Default" to match the above table. Also it makes the purpose clearer to me.

File: content/docs/user-guide/assets/asset-processor/settings.md [hunk 16]
```diff
@@ -0,0 +1,46 @@
+---
+linkTitle: Settings
+title: Asset Processor Settings
+description: Commandline settings for AssetProcessor and AssetBuilder
```
Comment (c5h11oh) [inline content/docs/user-guide/assets/asset-processor/settings.md:4]: I don't see settings for AssetBuilder in this page. Do we mean AssetProcessorBatch?

File: content/docs/user-guide/assets/asset-processor/settings.md (hunk 14 above)
Comment (lmbr-pip) [reply content/docs/user-guide/assets/asset-processor/settings.md:38]: Fixed. Copied typo from code comment. Nice catch

File: content/docs/user-guide/assets/asset-processor/settings.md (hunk 16 above)
Comment (lmbr-pip) [reply content/docs/user-guide/assets/asset-processor/settings.md:4]: Fixed. Good catch

File: content/docs/user-guide/assets/asset-processor/settings.md [hunk 17]
```diff
@@ -0,0 +1,46 @@
...
+linkTitle: Settings
+title: Asset Processor Settings
+description: Commandline settings for AssetProcessor and AssetProcessorBatch
+weight: 900
+toc: true
+---
+
+The Asset Processor and Asset Processor Batch expose a number of commandline settings that can be used when manually launching.
+
+For example, if you wanted to launch AssetProcessor manual on Windows but delay launch to attach a debugger:
```
Comment (micronAMZN) [inline content/docs/user-guide/assets/asset-processor/settings.md:11]: nit

```suggestion
For example, if you want to manually launch Asset Processor on Windows but delay launch to attach a debugger:
```

File: content/docs/user-guide/assets/asset-processor/settings.md [hunk 18]
```diff
@@ -0,0 +1,46 @@
...
+| additionalScanFolders           | Used with `dependencyScanPattern` to farther filter the scan.                                                                                                                                               |                                                                                                 |
+| debugOutput                     | When enabled, builders that support it will output debug information as product assets. Used primarily with scene files.                                                                                    |                                                                                                 |
+| dependencyScanMaxIteration      | Used to limit the number of recursive searches per line when running dependencyScanPattern.                                                                                                                 | Default is 800.                                                                                 |
```
Comment (micronAMZN) [inline content/docs/user-guide/assets/asset-processor/settings.md:35]:
```suggestion
| dependencyScanMaxIteration      | Used to limit the number of recursive searches per line when running dependencyScanPattern.                                                                                                                 | 800                                                                                 |
```

File: content/docs/user-guide/assets/asset-processor/settings.md [hunk 19]
```diff
@@ -0,0 +1,46 @@
...
+| sortJobsByDBSourceName          | When enabled, sorts pending jobs with equal priority and dependencies by database source name instead of job ID. Useful for automated tests to process assets in the same order each time.                  |                                                                                                 |
+| truncatefingerprint             | Truncates the fingerprint used for processed assets. Useful if you plan to compress product assets to share on another machine because some compression formats like zip will truncate file mod timestamps. |                                                                                                 |
+| waitOnLaunch                    | Briefly pauses Asset Processor during initialization. Useful if you want to attach a debugger.                                                                                                              | Defaults to off, if True then wait for 20s to aid debugger attach.                              |
```
Comment (micronAMZN) [inline content/docs/user-guide/assets/asset-processor/settings.md:42]:
```suggestion
| waitOnLaunch                    | Briefly pauses Asset Processor during initialization.  Waits for 20 seconds when true. Useful if you want to attach a debugger.                                                                                                              | false                             |
```

File: content/docs/user-guide/assets/asset-processor/settings.md [hunk 20]
```diff
@@ -0,0 +1,46 @@
...
+| warningLevel                    | Configure the error and warning reporting level for AssetProcessor. 0 Default, 1 for fatal errors, 2 for fatal errors and warnings.                                                                         | Defaults to 0.                                                                                  |
```
Comment (micronAMZN) [inline content/docs/user-guide/assets/asset-processor/settings.md:43]: Is 0 off? No warnings or fatal errors?

```suggestion
| warningLevel                    | Configure the error and warning reporting level for Asset Processor. 1 for fatal errors, 2 for fatal errors and warnings.                                                                         | 0                                                                                  |
```

File: content/docs/user-guide/assets/asset-processor/settings.md [hunk 21]
```diff
@@ -0,0 +1,46 @@
...
+| zeroAnalysisMode                | Enables using file modification time when examining source assets for processing.                                                                                                                           | AP defaults to true for new users, can be changed in UI for AP. Defaults to false for AP Batch. |
```
Comment (micronAMZN) [inline content/docs/user-guide/assets/asset-processor/settings.md:44]:
```suggestion
| zeroAnalysisMode                | Use file modification time when examining source assets for processing.                                                                                                                           | Asset Processor defaults to true, and can be changed in Asset Processor's UI. Asset Processor Batch defaults to false. | 
```

File: content/docs/user-guide/assets/asset-processor/settings.md [hunk 22]
```diff
@@ -0,0 +1,46 @@
...
+```
+
+## General settings
+When manually launching Asset Processor or Asset Processor Batch you can set the following general properties:
+
+| Setting            | Description                                                         | Default value                                               |
+|--------------------|---------------------------------------------------------------------|-------------------------------------------------------------|
+| project-name       | Name of the current project.                                        | The project name associated with the passed `project-path`. |
+| project-cache-path | Full path to the project cache folder.                              | The cache path associated with the passed `project-path`.   |
+| project-path       | Supplies the path to the project that the AP should use (required). |                                                             |
```
Comment (micronAMZN) [inline content/docs/user-guide/assets/asset-processor/settings.md:23]:
```suggestion
| project-path       | Supplies the path to the project that the Asset Processor should use (required). |                                                             |
```

**Review by micronAMZN — APPROVED** [review]
Approved with handful of minor editorial suggestions in one of the topics.

File: content/docs/user-guide/assets/asset-processor/settings.md (hunk 20 above)
Comment (lmbr-pip) [reply content/docs/user-guide/assets/asset-processor/settings.md:43]: No 0 is accept the default settings for logging for O3DE, which I believe is log everything at Info level.

**Pull request merged**